code/edgeconsent/src/test/java/com/adobe/marketing/mobile/edge/consent/ConsentTest.java

```diff
@@ -68,158 +74,195 @@ public void test_extensionVersionAPI() {
 	// registerExtension
 	// ========================================================================================
 	@Test
-	public void testRegistration() {
-		// test
-		Consent.registerExtension();
-		final ArgumentCaptor<ExtensionErrorCallback> callbackCaptor = ArgumentCaptor.forClass(
-			ExtensionErrorCallback.class
-		);
+	public void testRegistration_deprecated() {
+		try (MockedStatic<MobileCore> mobileCoreMockedStatic = Mockito.mockStatic(MobileCore.class)) {
+			// mock MobileCore.registerExtension()
+			ArgumentCaptor<Class> extensionClassCaptor = ArgumentCaptor.forClass(Class.class);
+			ArgumentCaptor<ExtensionErrorCallback> callbackCaptor = ArgumentCaptor.forClass(
+				ExtensionErrorCallback.class
+			);
+			mobileCoreMockedStatic
+				.when(() -> MobileCore.registerExtension(extensionClassCaptor.capture(), callbackCaptor.capture()))
+				.thenReturn(true);
+			// call registerExtension() API
+			Consent.registerExtension();
+			// verify: happy
+			assertNotNull(callbackCaptor.getValue());
+			assertEquals(ConsentExtension.class, extensionClassCaptor.getValue());
+			// verify: not exception when error callback was called
+			callbackCaptor.getValue().error(ExtensionError.UNEXPECTED_ERROR);
+		}
+	}
 
-		// The consent extension should register with core
-		PowerMockito.verifyStatic(MobileCore.class, Mockito.times(1));
-		MobileCore.registerExtension(ArgumentMatchers.eq(ConsentExtension.class), callbackCaptor.capture());
+	// ========================================================================================
+	// publicExtensionConstants
+	// ========================================================================================
+	@Test
+	public void test_publicExtensionConstants() throws InterruptedException {
```

Comment:
nit: IMO this test is more relevant in functional tests. Latches and thread waits make unit tests flaky.